cli_tools/common/utils/logging/logger.go

```diff
@@ -15,50 +15,205 @@
 package logging
 
 import (
-	"fmt"
 	"log"
+	"os"
+	"strings"
+	"sync"
 	"time"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/proto/go/pb"
+	"github.com/golang/protobuf/proto"
 )
 
-// LogEntry encapsulates a single log entry.
-type LogEntry struct {
-	LocalTimestamp time.Time `json:"localTimestamp"`
-	Message        string    `json:"message"`
+// LogWriter is a logger for interactive tools. It supports
+// string messages and structured metrics.
+//
+// Structured metrics are accumulated over the lifespan of the logger.
+//
+// To rebuild the mock, run `go generate ./...`
+//go:generate go run github.com/golang/mock/mockgen -package mocks -source $GOFILE -destination ../../../mocks/mock_logger.go
+type LogWriter interface {
+	// WriteUser messages appear in the following places:
+	//  1. Web UI and gcloud.
+	//  2. Standard output of the CLI tool.
+	//  3. Backend trace logs (all debug and user logs are combined to a single trace log).
+	WriteUser(message string)
```

Comment:
Consider using Log* prefix or no prefix at all. Eg. Info(), Debug(), Trace()

Reply:
Changed to no prefix